py2app/recipes/docutils.py

```diff
@@ -1,14 +0,0 @@
-def check(cmd, mf):
-    m = mf.findNode("docutils")
-    if m is None or m.filename is None:
-        return None
-    for pkg in [
-        "languages",
-        "parsers",
-        "readers",
-        "writers",
-        "parsers.rst.directives",
-        "parsers.rst.languages",
-    ]:
-        mf.import_hook("docutils." + pkg, m, ["*"])
-    return {}
```

Comment:
A disadvantage of switching to autopackages is that docutils will no longer be included in site-packages.zip.  I'd prefer to keep this recipe instead of moving to autopackages.

Reply:
I'd prefer to keep it in the zip directory also. Unfortunately, it is not just references to `__file__` that cause path errors. In the case of `docutils`, `[Errno 20] Not a directory: '...Contents/Resources/lib/python39.zip/docutils/writers/latex2e/docuttils.sty'` results. I can investigate to see exactly what is the source of the error, perhaps `os.path.exists` (see comment below)

Reply:
In the circumstance that I encountered, `docutils.writers.latex2e.__init__.py` line 575 references `__file__`. I suppose that py2app missed this. Perhaps all the writers (and readers and parsers?) in docutils need to be included, not just the writer (and readers and parsers) module. But then, I suppose that py2app would detect the `__file__` reference and put docutils in the packages list anyway?